lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4787,6 +4787,34 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           }
         }
 
+        // Require the superclass to have an available designated initializer,
+        // unless we have already determined the class or superclass is invalid.
+        if (!isInvalidSuperclass && !Super->isInvalid()) {
+          bool superIsConstructible = false; // can be satisfied by convenience initializers
```

Comment:
Another nit: all comments are capitalized and punctuated, and this file respects the 80-character limit for each line.